metricbeat/module/docker/memory/data.go

```diff
@@ -29,20 +29,31 @@ func eventsMapping(r mb.ReporterV2, memoryDataList []MemoryData) {
 }
 
 func eventMapping(r mb.ReporterV2, memoryData *MemoryData) {
-	fields := common.MapStr{
-		"fail": common.MapStr{
-			"count": memoryData.Failcnt,
-		},
-		"limit": memoryData.Limit,
-		"rss": common.MapStr{
-			"total": memoryData.TotalRss,
-			"pct":   memoryData.TotalRssP,
-		},
-		"usage": common.MapStr{
-			"total": memoryData.Usage,
-			"pct":   memoryData.UsageP,
-			"max":   memoryData.MaxUsage,
-		},
+
+	//if we have windows memory data, just report windows stats
+	var fields common.MapStr
+	if memoryData.Commit+memoryData.CommitPeak+memoryData.PrivateWorkingSet > 0 {
+		fields = common.MapStr{
+			"commit":              memoryData.Commit,
+			"commit_peak":         memoryData.CommitPeak,
+			"private_working_set": memoryData.PrivateWorkingSet,
```

Comment:
Follow the structure of Linux metrics, in case some day docker supports limits in Windows and we can calculate percentages.

Reply:
++